File: PerchReadyApp/server/java/com/ibm/mil/ready/app/perch/model/TipAction.java
/*
 * Licensed Materials - Property of IBM © Copyright IBM Corporation 2015. All
 * Rights Reserved. This sample program is provided AS IS and may be used,
 * executed, copied and modified without royalty payment by customer (a) for its
 * own instruction and study, (b) in order to develop applications designed to
 * run with an IBM product, either for customer's own internal use or for
 * redistribution by customer, as part of such an application, in customer's own
 * products.
 */

/**
 *
 */
package com.ibm.mil.ready.app.perch.model;

/**
 * @author tannerpreiss
 *
 */
public class TipAction {
	private String title;
	private String message;
	private String iconName;
	private String type;

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

	public String getIconName() {
		return iconName;
	}

	public void setIconName(String iconName) {
		this.iconName = iconName;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

}

File: PerchReadyApp/apps/Perch/iphone/native/Perch/Models/TipAction.swift
/*
Licensed Materials - Property of IBM
© Copyright IBM Corporation 2015. All Rights Reserved.
*/

import UIKit

/**
*  Data model to hold Incentive or Product data on a Tip object, if available
*/
class TipAction: JsonObject {
    
    var iconName: String!
    var message: String!
    var title: String!
    var type: String!
    
    /// Computed icon name based on mapping
    var actualIconName: String {
        return TipDataManager.sharedInstance.incentiveImageMapping(iconName)
    }
   
}
